src/reactapp/src/components/checkoutAgreements/CheckoutAgreementsMemorized.jsx

```diff
@@ -0,0 +1,41 @@
+import React, { useState } from 'react';
+
+import Card from '../common/Card';
+import Header from '../common/Header';
+import CheckoutAgreementsForm from './components/CheckoutAgreementsForm';
+import CheckoutAgreementModal from './components/CheckoutAgreementModal';
+import CheckoutAgreementFormikProvider from './components/CheckoutAgreementsFormikProvider';
+import { __ } from '../../i18n';
+import { _isObjEmpty } from '../../utils';
+import { formikDataShape } from '../../utils/propTypes';
+import useAgreementAppContext from './hooks/useAgreementAppContext';
+
+const CheckoutAgreementsMemorized = React.memo(({ formikData }) => {
+  const [activeModalId, setActiveModalId] = useState(false);
+  const { checkoutAgreements } = useAgreementAppContext();
+
+  return (
+    <CheckoutAgreementFormikProvider formikData={formikData}>
+      {_isObjEmpty(checkoutAgreements) ? (
```

Comment:
Is it important to do this check inside that provider? Otherwise I'd normally prefer to exit as soon as possible. It makes for cleaner markup:

```
  if _isObjEmpty(checkoutAgreements) {
    return '';
  }

  return (
    <CheckoutAgreementFormikProvider formikData={formikData}>
      <Card>
      ....
```

Reply:
We want `<CheckoutAgreementFormikProvider formikData={formikData}>` always present here. Because this component registers the agreement formik section to the application. Also, there is where we fetch the agreements.

Reply:
thanks for explaining!